backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/IssueMatchingService.java

```diff
@@ -204,9 +204,12 @@ private void processEvent(AnalysisFinishedEvent event) {
           }
           var localIssueMatcher = new IssueMatcher<>(new KnownIssueMatchingAttributesMapper(), new TrackedIssueFindingMatchingAttributeMapper());
           var localMatchingResult = localIssueMatcher.match(previouslyKnownIssues, e.getValue());
-          return localMatchingResult.getMatchedLefts().entrySet().stream()
-            .map(matchedEntry -> updateTrackedIssueWithPreviousTrackingData(matchedEntry.getKey(), matchedEntry.getValue()))
-            .collect(toList());
+          var matchedAndUnmatchedIssues = localMatchingResult.getMatchedLefts().entrySet().stream()
+            .map(matchedEntry -> updateTrackedIssueWithPreviousTrackingData(matchedEntry.getKey(), matchedEntry.getValue())).collect(Collectors.toCollection(ArrayList::new));
+          List<TrackedIssue> unmatchedIssues =
+            StreamSupport.stream(localMatchingResult.getUnmatchedRights().spliterator(), false).collect(Collectors.toList());
+          matchedAndUnmatchedIssues.addAll(unmatchedIssues);
+          return matchedAndUnmatchedIssues;
```

Comment:
```suggestion
          var matchedIssues = localMatchingResult.getMatchedLefts().entrySet().stream()
            .map(matchedEntry -> updateTrackedIssueWithPreviousTrackingData(matchedEntry.getKey(), matchedEntry.getValue()));
          var unmatchedIssues = StreamSupport.stream(localMatchingResult.getUnmatchedRights().spliterator(), false);
          return Stream.concat(matchedIssues, unmatchedIssues).collect(toList());
```
It would be more readable if you could name the matched and unmatched issues and combine them at the end.